File: .tmp_py4013/batch_11_dump/3218_minimum_cost_for_cutting_cake_i.js
// LeetCode 3218 - Minimum Cost for Cutting Cake I
// https://leetcode.com/problems/minimum-cost-for-cutting-cake-i/

var minimumCost = function(m, n, horizontalCut, verticalCut) {
    horizontalCut.sort((a, b) => b - a);
    verticalCut.sort((a, b) => b - a);
    let i = 0, j = 0, h = 1, v = 1, ans = 0;
    while (i < m - 1 || j < n - 1) {
        if (j === n - 1 || (i < m - 1 && horizontalCut[i] > verticalCut[j])) {
            ans += horizontalCut[i] * v;
            h++; i++;
        } else {
            ans += verticalCut[j] * h;
            v++; j++;
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check15/3218_minimum_cost_for_cutting_cake_i.rs
struct Solution;
// LeetCode 3218 - Minimum Cost for Cutting Cake I
// https://leetcode.com/problems/minimum-cost-for-cutting-cake-i/

impl Solution {
    pub fn minimum_cost(m: i32, n: i32, mut horizontal_cut: Vec<i32>, mut vertical_cut: Vec<i32>) -> i32 {
        horizontal_cut.sort_unstable_by(|a, b| b.cmp(a));
        vertical_cut.sort_unstable_by(|a, b| b.cmp(a));
        let mut i = 0;
        let mut j = 0;
        let mut h = 1;
        let mut v = 1;
        let mut ans = 0;
        while i < m - 1 || j < n - 1 {
            if j == n - 1 || (i < m - 1 && horizontal_cut[i as usize] > vertical_cut[j as usize]) {
                ans += horizontal_cut[i as usize] * v;
                h += 1;
                i += 1;
            } else {
                ans += vertical_cut[j as usize] * h;
                v += 1;
                j += 1;
            }
        }
        ans
    }
}

fn main() {}
